### PR TITLE
submit clears redux state

### PR DESCRIPTION
**Pull request by abrande**
- Added action to clear redux state
- All reducers return INITIAL_STATE when reset state is invoked
- ReduxFormClear is only called after successful user creation
- SubmissionModal now renders with native 'open' property
- Tests for all

### CODE REVIEW

**Review by vaibhavdubey08 — APPROVED** [review]

File: app/components/SignUp/UserDetails/UserInfoForm.js [hunk 1]
```diff
@@ -19,7 +20,7 @@ export class UserInfoForm extends React.Component {
   };
 
   closeModal = () => {
-    this.props.clearResult();
+    this.props.resetReduxState();
```
Comment (nicko-winner) [inline app/components/SignUp/UserDetails/UserInfoForm.js:23]: Is this modal being show always on success and on failure? if it's being called on failure do, do we want to reset the state? Hypothetical use case: User enters something wrong, gets an error, maybe they want to go back a couple pages and try to fix it?

Comment (abrande) [reply app/components/SignUp/UserDetails/UserInfoForm.js]: Has been fixed to only clear on successful user creation

File: app/components/SignUp/UserDetails/UserInfoForm.js [hunk 2]
```diff
@@ -52,7 +60,7 @@ export class UserInfoForm extends React.Component {
             </div>
           </div>
         </form>
-        <SubmissionModal openModal={result} hideModal={this.closeModal} />
+        {result && <SubmissionModal open={this.state.open} hideModal={this.closeModal} />}
```
Comment (abrande) [inline app/components/SignUp/UserDetails/UserInfoForm.js:63]: Result is always present after an initial submission. This caused a bug if the user receives an error. The modal would automatically appear on the page where the error occurs. In this case the user is unable to change the input, and is navigated back to the first page after closing the modal. Now the modal **only** appears after clicking the submit button.

File: app/actions/resetReduxState.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+// This is used for all redux states to return to the original state upon submission
```
Comment (nicko-winner) [inline app/actions/resetReduxState.js:1]:
```suggestion
// This is used for all redux reducers to return to the original state upon submission
```

**Review by vaibhavdubey08 — APPROVED** [review]

**Review by nicko-winner — APPROVED** [review]

**Pull request merged**